File: pre-2014-06/inaimathi/aromatic-numbers/aromatic.erl
-module(aromatic).
-compile(export_all).

lookup(73) -> 1;	%% I
lookup(86) -> 5;	%% V
lookup(88) -> 10;	%% X
lookup(76) -> 50;	%% L
lookup(67) -> 100;	%% C
lookup(68) -> 500;	%% D
lookup(77) -> 1000.	%% M

fromAromatic(List) -> fromAromatic(List, 0).
fromAromatic([A, R, A2, R2 | Rest], Sum) ->
    Prod = (A-48)*lookup(R), Next = [A2, R2 | Rest],
    case R2 > R of
	true -> fromAromatic(Next, Sum - Prod);
	false -> fromAromatic(Next, Sum + Prod)
    end;
fromAromatic([A, R], Sum) ->
    Sum + ((A-48)*lookup(R)).

File: older-code/inaimathi/aromatic-numbers/aromatic.hs
import Data.Maybe
import Data.Char

lookRoman char = fromJust $ lookup char charTable
  where charTable = zip "IVXLCDM" [1, 5, 10, 50, 100, 500, 1000]

fromAromatic :: String -> Int
fromAromatic aromatic = rec aromatic 0
  where rec (a:r:a':r':rest) total = rec (a':r':rest) $ if r' > r then total - (prod a r) else total + (prod a r)
        rec (a:r:[]) total = total + (prod a r)
        prod a r = (digitToInt a) * (lookRoman r)